### PR TITLE
New Caption Search Controller

### PR DESCRIPTION
**Pull request by wenyugu**

### CODE REVIEW

File: ClassTranscribeServer/ClassTranscribeServer.csproj [hunk 1]
```diff
@@ -71,6 +71,8 @@
     <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="5.0.0" />
     <PackageReference Include="Swashbuckle.AspNetCore.SwaggerGen" Version="5.0.0" />
     <PackageReference Include="Swashbuckle.AspNetCore.SwaggerUI" Version="5.0.0" />
+    <PackageReference Include="NEST" Version="7.9.0" />
```
Comment (angrave) [inline ClassTranscribeServer/ClassTranscribeServer.csproj:74]: 7.10 has been released
https://github.com/elastic/elasticsearch-net/releases

File: ClassTranscribeServer/ClassTranscribeServer.csproj [hunk 2]
```diff
@@ -71,6 +71,8 @@
...
+    <PackageReference Include="Elasticsearch.Net" Version="7.9.0" />
```
Comment (angrave) [inline ClassTranscribeServer/ClassTranscribeServer.csproj:75]: (If 7.10 is preferable, then both of these should be updated)

File: ClassTranscribeServer/Controllers/CaptionsSearchController.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        public CaptionsSearchController(CTDbContext context,
+            IElasticClient client,
+            ILogger<CaptionsSearchController> logger) : base(context, logger)
+        {
+            _elasticClient = client;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<Caption>> Search(string index, string keyword)
```
Comment (angrave) [inline ClassTranscribeServer/Controllers/CaptionsSearchController.cs]: Todo: Will we need a security access check implemented here?

File: ClassTranscribeServer/Startup.cs [hunk 4]
```diff
@@ -150,7 +151,10 @@ public void ConfigureServices(IServiceCollection services)
             services.AddScoped<UserUtils>();
             services.AddScoped<CaptionQueries>();
 
-            var settings = new ConnectionSettings(new Uri("http://localhost:9200/"));
+            // Configure ElasticSearch client
+            var configuration = CTDbContext.GetConfigurations();
+            var connection = new Uri(configuration.GetValue<string>("ES_CONNECTION_ADDR"));
```
Comment (angrave) [inline ClassTranscribeServer/Startup.cs:157]: Please add to AppSettings to be consistent with rest of code

File: ClassTranscribeServer/Controllers/CaptionsSearchController.cs (hunk 3 above)
Comment (wenyugu) [reply ClassTranscribeServer/Controllers/CaptionsSearchController.cs]: Yes, I think that will be implemented once we have everything working after the frontend is integrated. Would that be okay?

Comment (angrave) [reply ClassTranscribeServer/Controllers/CaptionsSearchController.cs]: Yes of course! Suggest you at least put a code comment in though "TODO:...."

Comment (wenyugu) [reply ClassTranscribeServer/Controllers/CaptionsSearchController.cs]: Done

**Comment by angrave** [comment]
Please make a new no-op commit to retrigger the new build scripts

**Comment by angrave** [comment]
If you prefer you could push or generate a PR into "expt" branch which will also autobuild

**Review by angrave — APPROVED** [review]

**Comment by angrave** [comment]
Do you want to merge or shall I?

**Pull request merged**